node/forest_libp2p/src/service.rs

```diff
@@ -172,6 +187,28 @@ impl Libp2pService {
                                 response,
                             }).await;
                         }
+                        ForestBehaviourEvent::BitswapReceivedBlock(peer_id, cid, block) => {
+                            match self.db.put(&block, Blake2b256) {
+                                Ok(actual_cid) => {
+                                    trace!("saved bitswap block with returned cid {:?}, expected cid: {:?}", actual_cid, cid);
+                                }
+                                Err(e) => {
+                                    warn!("failed to save bitswap block: {:?}", e.to_string());
+                                }
```

Comment:
Question(s): 
- Why check if the actual is different from the expected without any action if that is true?
- Why do we need to `put` the bitswap blocks here? Wouldn't they be written to the DB during sync? I understand that we need to store the block to be able to send out `ReceivedWant` blocks but I figured it would be stored during the syncing process after validation?

Reply:
I dont check for it. I just log the 2 Cids for debugging purposes. I am not entirely sure how we want to handle scenarios like this yet... 

I'd say its okay to save messages to the db before checking them. Under the assumtion that the Cid of the data matches what we get for the Cid when we save it, we can delete it after. This design might change though. This PR doesnt involve using bitswap yet, so some of this stuff will probably change.